File: Java/prog_lab_7/src/prog_lab_7/figure.java
package prog_lab_7;

public abstract class figure {//абстрактный класс
	protected  point[] pt;	
	figure(int n){
	pt = new point[n];
		for(int i = 0; i < n; i++) {
			pt[i] = new point();
		}
	}
	abstract double square();
	abstract double per();
}

File: C#/prog_lab_6/figure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace prog_lab_6
{
	internal abstract class figure
	{
		protected point[] pt;
		protected figure(int n)
		{
			pt = new point[n];
			for (int i = 0; i < n; i++)
			{
				pt[i] = new point();
			}
		}
		abstract public double square();
		abstract public double per();
	}
}
